echo_test.go

```diff
@@ -17,11 +17,127 @@
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
 	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
 )
 
 const prefixURI = "/v0"
 
 func TestInvalidMethod(t *testing.T) {
 
+	req, err := http.NewRequest("GET", "/v0/echo", nil)
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	rr := httptest.NewRecorder()
+	handler := EchoHandler{}
+	handler.ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusMethodNotAllowed, rr.Code, "Incorrect status code returned")
+
+}
+
+func TestEmptyBody(t *testing.T) {
+
+	req, err := http.NewRequest("POST", "/v0/echo", strings.NewReader(""))
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	rr := httptest.NewRecorder()
+	handler := EchoHandler{}
+
+	handler.ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusBadRequest, rr.Code, "Incorrect status code returned")
+}
+
+func TestInvalidJSON(t *testing.T) {
+
+	req, err := http.NewRequest("POST", "/v0/echo", strings.NewReader("bad req body"))
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	rr := httptest.NewRecorder()
+	handler := EchoHandler{}
+
+	handler.ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusBadRequest, rr.Code, "Incorrect status code returned")
+}
+
+type errReader int
+
+func (errReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("test error")
+}
+
+func TestFailedBody(t *testing.T) {
+
+	req, err := http.NewRequest("POST", "/v0/echo", errReader(0))
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	rr := httptest.NewRecorder()
+
+	handler := EchoHandler{}
+
+	handler.ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusBadRequest, rr.Code, "Incorrect status code returned")
+}
+
+func TestTimeStamp(t *testing.T) {
+	msg, _ := json.Marshal(Echo{Message: "echo"})
+
+	req, erra := http.NewRequest("POST", "/v0/echo", bytes.NewBuffer(msg))
```

Comment:
You _can_ give each of these errors a different name (i.e. create a new error instance each time) if it helps to (for example) retain the state from an earlier error for later use.  That doesn't seem to be the case, so I'd just reuse them.

Now, when you use the `:=` operator, you're both _declaring_ a new variable and _assigning_ it a value (while telling the compiler to infer the type, in this case, whatever `http.NewRequest(...)` returns).  If you try to reuse it below (line 131, which is currently `errb`), and you use the `:=` operator, this will fail because you cannot re-declare a variable.  So to reuse it, use the `=` operator so that you're only reassigning the value, not re-declaring its type.